runtime/commonMain/src/kotlinx/serialization/json/internal/JsonReader.kt

```diff
@@ -165,94 +164,86 @@ internal class JsonReader(private val source: String) {
 
     fun nextToken() {
         val source = source
-        var curPos = currentPosition
-        val maxLen = source.length
-        while (true) {
-            if (curPos >= maxLen) {
-                tokenPosition = curPos
-                tokenClass = TC_EOF
-                return
-            }
-            val ch = source[curPos]
-            val tc = charToTokenClass(ch)
-            when (tc) {
-                TC_WS -> curPos++ // skip whitespace
+        var currentPosition = currentPosition
+        while (currentPosition < source.length) {
+            val ch = source[currentPosition]
+            when (val tc = charToTokenClass(ch)) {
+                TC_WS -> currentPosition++ // skip whitespace
                 TC_OTHER -> {
-                    nextLiteral(source, curPos)
+                    nextLiteral(source, currentPosition)
                     return
                 }
                 TC_STRING -> {
-                    nextString(source, curPos)
+                    nextString(source, currentPosition)
                     return
                 }
                 else -> {
-                    this.tokenPosition = curPos
+                    this.tokenPosition = currentPosition
                     this.tokenClass = tc
-                    this.currentPosition = curPos + 1
+                    this.currentPosition = currentPosition + 1
                     return
                 }
             }
         }
+
+        tokenPosition = currentPosition
+        tokenClass = TC_EOF
     }
 
     private fun nextLiteral(source: String, startPos: Int) {
         tokenPosition = startPos
         offset = startPos
-        var curPos = startPos
-        val maxLen = source.length
-        while (true) {
-            curPos++
-            if (curPos >= maxLen || charToTokenClass(source[curPos]) != TC_OTHER) break
+        var currentPosition = startPos
+        while (currentPosition < source.length && charToTokenClass(source[currentPosition]) == TC_OTHER) {
```

Comment:
Will `source.length` be optimized down to the local var to avoid getter call how it was with `val maxLen = source.length`?

Reply:
Yes, it will.
Note that such low-level optimizations should be profile-guided in the first place